### ISSUE TITLE
problem sets and learned material - timing 

### ISSUE THREAD

**Issue by emdavidson** [paultopia/gobbledygook#5]
So the single most pain that I had for the class is the fact that we had problem sets that would take more than a day or so to complete but required the entire lesson from that same week. I would strongly recommend either allowing student to turn in the homework on Monday or the following week. Other computer science courses that I have been in have finished all the material and that for say Problem Set 1, in say Week 4. Then it allowed time to digest, integrate, and apply the material learned from Week 4 during the Week 5, at the end of which the Problem Set 1 would be due. My brain does not work quickly at learning and being able to apply significantly different material and I know if there was a little more time for me to process and actually gain the skills in my mind, I would have done better and the problem sets would have been more fun and rewarding.

**Comment by rrectr** [comment]
I agree with this, it makes it much harder to attempt to manage your time with regards to other classes when there are questions that you cannot answer because we have not covered the material yet.  Additionally judging by problem set 3 where we delayed it to the Monday I think that did actually work better.

**Comment by amichel1007** [comment]
I agree with Emily, for a couple of the problem sets the crux of some of the hardest problems/solutions involved synthesizing material learned earlier in the week. That led to at least a couple of issues. 1) It seemed that we would be rushed the last week before a problem set was due to squeeze in critical material (understandable considering it's the first time running through the material) but that meant spending less time on some of the most important content, which led to issue 2) that for the round-table questions that were held before the problem sets were due, we wouldn't have time to review the necessary material to ask the questions that were the most difficult to figure out. As Emily said, extending the deadline until Sunday or Monday may help because no matter how much preparation was put in before the last week, the brunt of the problem set had to be completed between Tuesday after the last class and Friday at 5.
